src/actinia_core/core/stac_exporter_interface.py

```diff
@@ -0,0 +1,255 @@
+
+        result_catalog_validation = redis_actinia_interface.exists("result-catalog")
+
+        if result_catalog_validation is not TRUE:
+            results = Catalog(id="result-catalog", description="STAC catalog")
+
+            redis_actinia_interface.create(
+                "result-catalog",
+                results.to_dict(),
+            )
+        else:
+            raise AsyncProcessTermination("result-catalog is already created")
+
+    def stac_builder(self, output_path: str = None, filename: str = None,
+                     output_type: str = None):
+        """
+        This function build the STAC ITEM and implement the following extension:
+            - Projection
+            - Raster
+        Parameter:
+            Input:
+                - output_path = Path to the source
+                - filename =  name of the source
+                - output_type =  type of object (raster, vector)
+        """
+        if output_type == "raster":
+
+            # Get parameters for STAC item
+            extra_values = self._get_raster_parameters(output_path)
+
+            # Checking if the input has WGS 84 CRS
+            geom, bbox_raster = self._get_wgs84_parameters(extra_values)
+
+            # Start building the Item
+            item = Item(id=f"STAC-result-{filename}",
+                        geometry=geom,
+                        bbox=bbox_raster,
+                        datetime=datetime.utcnow(),
+                        properties={
+                            "gds": extra_values["gds"]
+                            }
+                        )
+
+            # Adding Asset and Raster Ext
+            asset_ = Asset(
+                href=output_path,
+                title=filename
+            )
+
+            item.add_asset(
+                key='source',
+                asset=asset_
+            )
+
+            # Adding the Projection Extension
+            proj_ext = ProjectionItemExt(item)
+            proj_ext.apply(
+                    epsg=extra_values["crs"],
+                    geometry=extra_values["geometry"],
+                    shape=extra_values["shape"],
+                    bbox=extra_values["bbox"],
+                    transform=extra_values["transform"]
+            )
+
+            # Adding the Raster Extension
+            item = self._set_raster_extention(output_path, item)
+
+            # Adding the Processing Extension
+            item = self._set_processing_extention(item)
+
+            # Read catalog from REDIS
+            catalog_dict = redis_actinia_interface.read("result-catalog")
+
+            # Create Catalog Object
+            catalog = read_dict(catalog_dict)
+
+            # Add item to Catalog
+            catalog.add_item(item)
+
+            # Update redis catalog
+            self._update_catalog_redis(catalog)
+
+        # TODO
+        elif output_type == "vector":
+            raise AsyncProcessTermination("Still under developments")
```

Comment:
To not create too high expectations:
```suggestion
            raise AsyncProcessTermination("Not implemented yet.")
```